### PR TITLE
Feature/add unit test to editor

### PR DESCRIPTION
**Pull request by ponelat**
@shockey the wip so far of the new editor tests.
I wouldn't look too hard at the source code changes ( since they need to be cleaned up, and their changes made known ).
But any input from a casual glance, and feedback therein is greatly appreciated!

List:
- [X] Create a fake ace instance to test against
- [X] Introduce enzyme for testing compomenents
- [X] Introduce a "origin" field for editor state
- [x] Conclude testing of `origin` field end to end ( ie: test editor-container, etc )
- [x] Fix source code that breaks nodejs tests ( ie: webpack file hooks )
- [X] Introduce marker testing
- [ ] Conclude testing markers
- [x] Cleanup any dodgy code and prepare for proper PR

### CODE REVIEW

**Comment by ponelat** [comment]
yo @shockey this is good for a review.
tl;dr
- Adds mechanism to only load yaml into the editor when the origin != editor
- Adds a fakeAce thing to test ace stuff. 
- That's about it, oh and tests!

**Comment by ponelat** [comment]
@shockey looks like we need to fix the validation tests... when run in isolation, they're good.
I think its taking longer than the XXXms we set for it.
Thoughts?

**Comment by shockey** [comment]
@ponelat, I've stepped the delay up in the past (via [`process.env.ASYNC_TEST_DELAY`](https://github.com/shockey/swagger-editor/blob/master/test/plugins/validate-semantic/validate-helper.js#L8)) and haven't gotten any consistently good results. I think there may be some other gremlin: possibly something that isn't isolating itself between each instance of the system that's being created? 

Also, if timing _is_ the issue, maybe a more intricate trailing edge throttle mechanism for delaying the test resolution until _n_ ms after the last state mutation would be better than a static delay.

I've marked that down for further investigation soon (https://github.com/orgs/swagger-api/projects/1#card-10252668), but I recall you wrote most of the current async test harness - so if anything comes to your mind, ping me and let's try to get to the bottom of it 😄

**Comment by ponelat** [comment]
Yeah, could very well be a leaky environment. Will need to take a look, hopefully its solveable by just updating the helper method.

**Comment by shockey** [comment]
@ponelat, I've just merged https://github.com/swagger-api/swagger-editor/pull/1786 which uses throttle to resolve the validation test after the Redux state has settled down for a time. let's see where it gets us 😄

**Comment by ponelat** [comment]
Nice one @shockey !

File: src/plugins/editor/components/editor.jsx [hunk 1]
```diff
@@ -161,21 +166,22 @@ export default function makeEditor({ editorPluginsToRun }) {
       })
     }
 
-    updateYaml = () => {
-      if(this.editor.getValue() === this.yaml[0]) {
```
Comment (shockey) [inline src/plugins/editor/components/editor.jsx]: Any reason this check was dropped? I don't imagine it could hurt anything.

Comment (ponelat) [reply src/plugins/editor/components/editor.jsx]: The changes in this PR are all about improving this area. Ie: only update editor when the `origin` is different. We can include this check if you like, but this check is not the core logic behind our defence against bad re-renders
